compiler/src/formatting/format.re

```diff
@@ -3672,6 +3674,52 @@ and print_expression =
 
   expression_doc;
 }
+
+and is_grouped_access_expression = (expr: Parsetree.expression) => {
+  switch (expr.pexp_desc) {
+  | PExpConstant(_)
+  | PExpTuple(_)
+  | PExpId(_)
+  | PExpArrayGet(_)
+  | PExpArraySet(_)
+  | PExpRecordGet(_)
+  | PExpRecordSet(_)
+  | PExpRecord(_)
+  | PExpBlock(_)
+  | PExpArray(_) => false
+  | PExpApp(func, _) =>
+    let func_name = get_function_name(func);
+    func_name != list_cons;
```

Comment:
Regular applications are allowed on the left side, so the `PExpApp` case should always be `false` like the others.

Reply:
Oh, of course,  and that captures lists too.